### PR TITLE
doc DOTNET_JITMinOpts=1

### PR DESCRIPTION
**Pull request by danmoseley**
Document how to get the equivalent of `Suppress JIT optimization on module load` setting in VS.

@AndyAyersMS @BruceForstall is this right?

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/runtime-config/compilation.md](https://github.com/dotnet/docs/blob/885d443d355271da52d6eac55504a0164604ec46/docs/core/runtime-config/compilation.md) | [Runtime configuration options for compilation](https://review.learn.microsoft.com/en-us/dotnet/core/runtime-config/compilation?branch=pr-en-us-37856) |

### CODE REVIEW

**Comment by danmoseley** [comment]
cc @tommcdon

File: docs/core/runtime-config/compilation.md [hunk 1]
```diff
@@ -139,6 +139,17 @@ Project file:
 | - | - | - |
 | **Environment variable** | `COMPlus_ReadyToRun` or `DOTNET_ReadyToRun` | `1` - enabled<br/>`0` - disabled |
 
+## Debuggability
+
+- Configures whether the .NET Core runtime generates optimized code for images that were built in release configuration.
```
Comment (gewarren) [inline docs/core/runtime-config/compilation.md:144]:
```suggestion
- Configures whether the .NET runtime generates optimized code for images that were built in release configuration.
```

File: docs/core/runtime-config/compilation.md [hunk 2]
```diff
@@ -139,6 +139,17 @@ Project file:
...
+- If you omit this setting, .NET generates optimized code. This is equivalent to setting the value to `0`.
+- Use this setting to make it easier to debug code generated from images built in release configuration.
```
Comment (gewarren) [inline docs/core/runtime-config/compilation.md:146]:
```suggestion
- Set the value to 1 to make it easier to debug code generated from images built in release configuration.
```

File: docs/core/runtime-config/compilation.md [hunk 3]
```diff
@@ -139,6 +139,17 @@ Project file:
...
+- See also [JIT Optimization and Debugging](/visualstudio/debugger/jit-optimization-and-debugging) which describes the equivalent setting in Visual Studio.
```
Comment (gewarren) [inline docs/core/runtime-config/compilation.md:147]:
```suggestion
- [JIT Optimization and Debugging](/visualstudio/debugger/jit-optimization-and-debugging) describes the equivalent setting in Visual Studio.
```

**Review by gewarren — APPROVED** [review]

**Comment by danmoseley** [comment]
It looks like I need the code owners to agree this is OK to document.

**Comment by AndyAyersMS** [comment]
It's actually not quite the same as the debugger experience, though it's fairly close. The debugger experience is signaled to the JIT by the runtime via `CORJIT_FLAG_DEBUG_CODE` flag bit on the jit request.

There is a separate flag bit `CORJIT_FLAG_MIN_OPT` that tells the jit not to optimize. `CORJIT_FLAG_MIN_OPT` can be forcibly set via `DOTNET_JITMinOpts=1`.

Many but not all parts of the JIT treat these flags the same way, but there are some differences.

**Comment by danmoseley** [comment]
Are you comfortable documenting this, or would rather not? Happy to close if not. I do think there is value in such an env var, but if it's not in a stable state then we don't need to put in docs.

**Comment by BruceForstall** [comment]
I think we should _not_ document this.

Instead, I think if we think this is a valuable user experience addition, we should expose the existing DOTNET_JitDebuggable variable which is currently `_DEBUG` only. That would match the VS option precisely (I think).

**Comment by BruceForstall** [comment]
And maybe rename the variable `DOTNET_SuppressJitOptimizationForDebugging`?

**Comment by AndyAyersMS** [comment]
I agree with @BruceForstall here. `DOTNET_JITMinOpts=1` is something we occasionally ask customers to try when we suspect there might be an optimizer bug, but I'm not sure it has a lot of use beyond that, since we don't want to specify exactly what parts of the debug experience might be preserved or lost.

**Comment by danmoseley** [comment]
opened https://github.com/dotnet/runtime/issues/94280

and if/when that's done I can disinter this PR.

**Comment by BruceForstall** [comment]
Thanks @danmoseley for the PR and issue!

**Pull request closed without merging**